### PR TITLE
feat: add release-drafter

### PR DESCRIPTION
**Pull request by Revolyssup**
### Type of change:
Adding release drafter.
Reference - https://github.com/release-drafter/release-drafter

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [ ] Documentation
- [ ] Refactor
- [ ] Chore
- [ ] CI/CD or Tests

### Pre-submission checklist:

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Review by AlinsRan — APPROVED** [review]
Good.

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    label: 'area/dependencies'
+change-template: '- $TITLE @$AUTHOR (#$NUMBER)'
+template: |
+  ## What's New
+  $CHANGES
+
+  ## 👨🏽‍💻 Contributors
+
+  Thank you to our contributors for making this release possible:
+  $CONTRIBUTORS
```
Comment (shreemaan-abhishek) [inline .github/release-drafter.yml:40]:
```suggestion
  $CONTRIBUTORS
  
```

**Review by shreemaan-abhishek — CHANGES_REQUESTED** [review]

**Comment by shreemaan-abhishek** [comment]
there should also be a CI with this isn't it?

**Comment by Revolyssup** [comment]
@shreemaan-abhishek done

File: .github/workflows/release-drafter.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    if: github.repository == 'apache/apisix-ingress-controller'
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Drafting release
+        id: release_drafter
+        uses: release-drafter/release-drafter@v5
+        with:
+          config-name: release-drafter.yml
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (shreemaan-abhishek) [inline .github/workflows/release-drafter.yml:38]:
```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

**Review by shreemaan-abhishek — APPROVED** [review]

**Pull request merged**